daemon/compress.c

```diff
@@ -0,0 +1,75 @@
+
+#include <str.h>
+#include <zlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include "log.h"
+#include "udp_listener.h"
+#include "str.h"
+#include "compress.h"
+
+#define windowBits 15
+
+#define GZIP_ENCODING 16
+#define ENABLE_ZLIB_GZIP 32
+
+static void init_stream(z_stream *stream);
```

Comment:
This forward declaration is unnecessary, as we are just defining it in the next line.